python/GafferOSLUI/OSLObjectUI.py

```diff
@@ -277,6 +279,158 @@ def __addPlug( self, name, defaultData ) :
 
 		],
 
+		"source" : [
+
+			"description",
+			"""
+			The input scene which provides the locations to be referenced by the `sourceLocations`
+			plugs.
+			"""
+
+		],
+
+		"sourceLocations" : [
+
+			"description",
+			"""
+			Defines additional scene locations to be made accessible via the `pointcloud_search()`,
+			`pointcloud_get()` and `transform()` OSL functions.
+			""",
+
+			"layout:section", "Source Locations",
+			"plugValueWidget:type", "GafferUI.LayoutPlugValueWidget",
+
+			"layout:customWidget:header:widgetType", "GafferOSLUI.OSLObjectUI._SourceLocationsHeader",
+			"layout:customWidget:header:index", 0,
+
+			"layout:customWidget:footer:widgetType", "GafferOSLUI.OSLObjectUI._SourceLocationsFooter",
+			"layout:customWidget:footer:index", -1,
+
+		],
+
+		"sourceLocations.*" : [
+
+			"deletable", True,
+			"label", "",
+
+			"layout:activator:isEnabled", lambda plug : plug["enabled"].getValue(),
+
+		],
+
+		"sourceLocations.*.name" : [
+
+			"description",
+			"""
+			The name to give to the location. This is how it will be referred to from OSL
+			in the `pointcloud_search()`, `pointcloud_get()` and `transform()` functions.
+			""",
+
+			"label", "",
+			"layout:activator", "isEnabled",
+			"layout:width", GafferUI.PlugWidget.labelWidth(),
+
+		],
+
+		"sourceLocations.*.enabled" : [
+
+			"description",
+			"""
+			Enables the location for access in OSL.
+			""",
+
+			"label", "",
+			"boolPlugValueWidget:displayMode", "switch",
+
+		],
+
+		"sourceLocations.*.location" : [
+
+			"description",
+			"""
+			The location to be made accessibe from OSL. This must exist in the
+			`source` scene.
+			""",
+
+			"label", "",
+			"layout:activator", "isEnabled",
+			"plugValueWidget:type", "GafferSceneUI.ScenePathPlugValueWidget",
```

Comment:
Think we're missing `"scenePathPlugValueWidget:scene", "source"` here so browsing uses the scene from the "source" plug.

Reply:
Also fixed in https://github.com/GafferHQ/gaffer/pull/6470/commits/881061a98be066e76bf73d040662f79407a18b8b.